### PR TITLE
[OIL-190] IE10 e2e test upgrades 

### PR DESCRIPTION
**Pull request by phogel**
Fixing tests for IE10. CPC and POI tests fail because they rely on the backend which does not have the required polyfills yet.

### CODE REVIEW

File: src/scripts/core/core_config.js [hunk 1]
```diff
@@ -170,6 +170,10 @@ export function getLanguage() {
 }
 
 export function getLanguageFromLocale(localeVariantName) {
+  if(!localeVariantName) {
+    logError('localeId missing in locale');
+    localeVariantName = 'en';
```
Comment (phogel) [inline src/scripts/core/core_config.js]: @ltparis2018 can you confirm if it is OK to pass a default here? It does not break any test.

Comment (ltparis2018) [reply src/scripts/core/core_config.js]: Yes, this is ok. OIL layer's default language is english. But the error message needs an improvement I think. What is locale id in this context? Or locale? This method does not know anything about the origin of given localeVariantName. For example, this method is invoked during outdated cookie transformation that does not know a locale or a locale id.

File: package.json [hunk 2]
```diff
@@ -123,6 +124,7 @@
     "pretty": "2.0.0",
     "puppeteer": "1.4.0",
     "raw-loader": "0.5.1",
+    "regenerator-runtime": "^0.11.1",
```
Comment (CanardSauvage) [inline package.json]: Who uses generators?

File: src/demos/auto-hide.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
 <script id="oil-configuration" type="application/configuration">
       {
         "poi_group_name": "axelSpringerSe_01",
-        "timeout" : "2"
+        "timeout" : "3"
```
Comment (CanardSauvage) [inline src/demos/auto-hide.html]: What is this for? It makes the test wait 1s longer!

File: src/polyfill.js [hunk 4]
```diff
@@ -1,10 +1,28 @@
...
-    // eslint-disable-next-line global-require
-    window.Promise = require('core-js/fn/promise');
-  } else {
-    return null;
+/* eslint-disable global-require */
+export function generatePolyfills(){
+  if(window.Promise) {
+    return;
   }
+  require('core-js/modules/es6.object.assign');
```
Comment (CanardSauvage) [inline src/polyfill.js]: No way we need all this polyfills!

File: test/e2e/advanced_settings_custom_purposes_test.js [hunk 5]
```diff
@@ -22,7 +22,10 @@ module.exports = {
       .waitForElementVisible(OIL_ADVANCED_SETTINGS_WRAPPER, 1000, false)
       .pause(100)
       .waitForElementPresent(OIL_CUSTOM_PURPOSE_SLIDER, 100, false)
-      .assert.containsText('html', 'Custom Purpose 1', 'Checking custom purpose title')
+      .pause(100)
+      .useCss()
+      // deactivate because opening the page in IE10 and 9 the custom purposes, only the e2e test fails
```
Comment (CanardSauvage) [inline test/e2e/advanced_settings_custom_purposes_test.js]: You should add a FIXME and the ticket number of the Jira Ticket you created.

File: test/e2e/auto_hide_test.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ module.exports = {
   'OIL Layer should be auto-hidden after the correct amount of time': function (browser) {
     browser
       .pause(500)
-      .waitForElementVisible(OIL_LAYER, 500, false)
+      .waitForElementVisible(OIL_LAYER, 2000, false)
       .pause(2000)
       .useXpath().waitForElementNotPresent(OIL_LAYER, 500)
```
Comment (CanardSauvage) [inline test/e2e/auto_hide_test.js]: If the timeout is 3s, how can this ever work?!

File: package.json (hunk 2 above)
Comment (phogel) [reply package.json]: Tried to mimic what babel-polyfill does here, looking for a solution. Deleting

File: src/polyfill.js (hunk 4 above)
Comment (phogel) [reply src/polyfill.js]: Have reduced them to the absolute necessary

File: test/e2e/advanced_settings_custom_purposes_test.js (hunk 5 above)
Comment (phogel) [reply test/e2e/advanced_settings_custom_purposes_test.js]: 👍

File: etc/browserstackTest.sh [hunk 7]
```diff
@@ -1,11 +1,15 @@
 #/bin/bash
 
 run_test() {
-  echo "- Testing $1 -"
+  echo "..."
+  echo "-------------------------"
+  echo "試）TESTING $1"
```
Comment (CanardSauvage) [inline etc/browserstackTest.sh]: Chinese?!

File: src/polyfill.js (hunk 4 above)
Comment (CanardSauvage) [reply src/polyfill.js]: This is super strange:

oil.js 19.569 kB -> 19.43 kB  **it got smaller!!!**
hub.js 11.239 kB -> 17.496 kB  **okay, this has the polyfills**

File: etc/browserstackTest.sh (hunk 7 above)
Comment (phogel) [reply etc/browserstackTest.sh]: The Japanese I know but probably the same character in Chinese, too.
試す【ためす】= to attempt; to test; to try out​
I've put it in as an eye catcher

**Pull request merged**